GVFS/GVFS.Common/NuGetUpgrader/NuGetUpgrader.cs

```diff
@@ -326,6 +337,17 @@ public override bool TryRunInstaller(InstallActionWrapper installActionWrapper,
                         return false;
                     }
 
+                    if (!this.noVerify)
+                    {
+                        if (!this.nuGetFeed.VerifyPackage(this.DownloadedPackagePath, this.nuGetUpgraderConfig.CertificateFingerprint))
+                        {
+                            error = "Package signature validation failed. Check the upgrade logs for more details.";
+                            activity.RelatedError(error);
+                            this.fileSystem.DeleteFile(this.DownloadedPackagePath);
+                            return false;
+                        }
+                    }
```

Comment:
This code looks the same as the new code in `TryDownloadNewestVersion`, can it be moved to a helper method?